app/src/main/kotlin/co/netguru/baby/monitor/client/feature/recording/RecordingFileController.kt

```diff
@@ -1,49 +1,45 @@
-package co.netguru.baby.monitor.client.feature.voiceAnalysis
+package co.netguru.baby.monitor.client.feature.recording
 
 import android.content.Context
+import co.netguru.baby.monitor.client.feature.voiceAnalysis.AacRecorder
 import io.reactivex.Single
 import org.threeten.bp.LocalDateTime
 import org.threeten.bp.format.DateTimeFormatter
 import timber.log.Timber
 import java.io.File
 import java.nio.ByteBuffer
 import java.nio.ByteOrder
+import javax.inject.Inject
 
-@Suppress("MagicNumber")
-object WavFileGenerator {
+class RecordingFileController @Inject constructor(
+    private val applicationContext: Context
+) {
 
-    internal const val DIRECTORY_NAME = "recordings"
-    private const val DATE_PATTERN = "yyyy_MM_dd_HH_mm_ss"
-
-    private const val BYTES_IN_MEGABYTE = 1_048_576L
-    private const val AVAILABLE_MEGABYTES_FOR_RECORDINGS = 50
-    private const val AVAILABLE_SPACE = AVAILABLE_MEGABYTES_FOR_RECORDINGS * BYTES_IN_MEGABYTE
-    private const val SUBCHUNK_1_SIZE_PCM = 16
-    private const val RIFF_HEADER = "RIFF"
-    private const val SUBCHUNK_1_ID = "fmt "
-    private const val SUBCHUNK_2_ID = "data"
-    private const val SUBCHUNK_2_ID_AND_DESC_SIZE = 8
-    private const val CHUNK_AND_SUBCHUNK_1_DESC_SIZE = 4 + 8
-    private const val FORMAT = "WAVE"
-    private const val AUDIO_FORMAT_PCM: Short = 1
-    private const val BITS_PER_BYTE = 8
-    private const val HEADER_SIZE = 44
+    fun saveRecording(
+        rawData: ByteArray
+    ) =
+        saveAudio(
+            rawData,
+            AacRecorder.BITS_PER_SAMPLE,
+            AacRecorder.CHANNELS,
+            AacRecorder.SAMPLING_RATE,
+            UPLOAD_RECORDINGS_DIRECTORY
+        )
 
-    fun saveAudio(
-        context: Context,
+    private fun saveAudio(
         rawData: ByteArray,
         bitsPerSample: Int,
         channels: Int,
-        sampleRate: Int
+        sampleRate: Int,
+        directoryName: String
     ) = Single.fromCallable {
+        // TODO add metadata to the recordings (should be consulted with ML dev)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/7c069c26782df05dac8a097aa956e351ad95f920/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/recording/RecordingFileController.kt#L36"></span>This comment contains text that has been defined as forbidden in detekt.</td>
    </tr>
  </tbody>
</table>